### PR TITLE
refactor: replace TRUE/FALSE macros with C99 true/false

### PR DESCRIPTION
**Pull request by dundargoc**
This is a work in progress to replace the TRUE/FALSE macro with the C99 true/false. This is in response to https://github.com/neovim/neovim/issues/459.

### CODE REVIEW

**Comment by dundargoc** [comment]
Dafuq happened? I just made an empty pull request and a bunch of other commits followed.

**Comment by janlazo** [comment]
https://github.com/neovim/neovim/wiki/Refactoring#frozen-legacy-modules

**Comment by dundargoc** [comment]
@janlazo Thank you for the link, I completely missed that part. I guess this leaves us with two choices:

1. Replace TRUE/FALSE macros everywhere except for `regexp.c`, `regexp_nfa.c` and `indent_c.c` and wait until they're replaced/deprecated before completely removing the TRUE/FALSE macros.

2. Make the absolutely bare minimum changes to the three files, i.e. only a string replace of TRUE-->true and FALSE-->false and nothing else.

I'd personally opt for number 2 since it doesn't technically break the significantly part of "should not be changed _significantly_" but I guess it's a question of interpretation. How would you like me to handle the three files?

**Comment by janlazo** [comment]
Option 2 is fine.

**Comment by janlazo** [comment]
First commit is empty. Avoid empty commits unless the commit message is for N/A vim patches.

**Comment by dundargoc** [comment]
@janlazo Thanks for the heads up, I'll fix it.

**Comment by janlazo** [comment]
Forgot to mention. Just because a variable is set to true/false doesn't imply that the variable is boolean. Every usage of the variable must be in the context of a boolean before changing the variable type from int to bool.

**Comment by dundargoc** [comment]
> Forgot to mention. Just because a variable is set to true/false doesn't imply that the variable is boolean. 
That seems a bit counter-intuitive. Wouldn't it make more sense to use true/false for booleans and 1/0 for integers as to not mix them together?

> Every usage of the variable must be in the context of a boolean before changing the variable type from int to bool.
OK. I interpret this to mean to only change variable type from int to bool if it's explicitly used in an if-statement?

**Comment by dundargoc** [comment]
@janlazo Or do you think I should skip changing variable types from int to bool for now and only focus on changing the macros?

**Comment by janlazo** [comment]
> > Forgot to mention. Just because a variable is set to true/false doesn't imply that the variable is boolean.
> 
> That seems a bit counter-intuitive. Wouldn't it make more sense to use true/false for booleans and 1/0 for integers as to not mix them together?
> 
> > Every usage of the variable must be in the context of a boolean before changing the variable type from int to bool.
> 
> OK. I interpret this to mean to only change variable type from int to bool if it's explicitly used in an if-statement?

That's what Neovim inherited and what I observed in some vim patches. Of course, it makes sense to not mix them up.

```c
if (foo) {
  bar();
}
if (!foobar) {
  do_foobar();
}
```
In this example, `foo` and `foobar` can be `bool` variables and it's safe to refactor if they're local variables or function parameters.

```c
if (*foo) {
  bar();
}
if (!(*foobar)) {
  do_foobar();
}
```
In this example, it is risky to refactor int pointers into boolean pointers, especially function parameters and global variables. I encountered test failures when refactoring int pointers.

File: src/clint.py [hunk 1]
```diff
@@ -3272,13 +3272,6 @@ def CheckLanguage(filename, clean_lines, linenum, file_extension,
                   " named ('k' followed by CamelCase) compile-time constant for"
                   " the size.")
 
-    # Detect TRUE and FALSE.
-    match = Search(r'\b(TRUE|FALSE)\b', line)
-    if match:
```
Comment (janlazo) [inline src/clint.py:3277]: Don't remove this beause it may resurface in future patches.

**Comment by dundargoc** [comment]
@janlazo I've removed the changes to clint. Should I squash all commits or make a commit for each type of change (aka one for only TRUE/FALSE->true/false and one for int->bool)?

**Comment by dundargoc** [comment]
Electing to close this since I'm not actively working on this particular PR and because all TRUE/FALSE macros will eventually be phased out regardless anyway.

**Pull request closed without merging**